README.md
# EDIBO
EDIBO projekta elektroniskā klāde
## Day 01 - Day 02
### Topics:
- Terminal (hot-keys)
- Shell (basics)
- Git (basics)
- ASCII table

#### Terminal (hot-keys)
Ctrl+Alt+T - open **Terminal**  
Ctrl+Shift+T - open a new tab in **Terminal**  
Ctrl+Shift+Q or ALT+F4 - close **Terminal**  
Ctrl + - Zoom In **Terminal**  
Ctrl - - Zoom Out **Terminal**  
Arrow Up and Arrow Down - List last commands  
Tab - Continue a command  

#### Shell (basics)
PS1="$ " - Rename first line in Shell for $
> It will not be written with a full name of PC, for example "user@333k-34:~$"

echo - First command to show (tell me)  
echo Hello World - Shows a text "Hello world"
echo $a - Shows what is a=?  
echo $(($a+$b)) - Summing  
echo $0 - Shows a name of a console  
pwd - Shows where are we  
cd "name" - Change directory  
cd .. - Change directory going upper
cd /home - Change directory "home"  
ls - List directory contents  
ls -l - List of directories with a time  
ls -a - Show all, include hidden  
ls -lt - List of directories with a last update
mkdir - Make a new directory  
rmdir - Remove a directory  
cat, tail, less, more - Shows a.txt (if made)  
date - Shows a time  
cal - Shows a calendar
sh - Another shell (not bash now)  
whoami - Command shows us who are we now  
who - Shows when PC was enabled  
last - We can see who is close  
tree - Shows all directory content

#### ASCII Table

[ASCII table](http://www.ecowin.org/ascii.htm)

## Day 04  

### Topics:  

- Darbs ar mainīgājiem  
- Darbs ar speciāliem mainīgājiem  
- Darbs ar mainīgo masīvu
- Darbs ar operācijām
- Darbs ar nosacījumu operatoru
- Darbs ar cikliem  

#### Darbs ar mainīgājiem  

#!/bin/bash  

NAME="Zara Ali"  
echo $NAME  

#### RESULT:

>Zara Ali  
---------------------------------  
Making a read-only:  

#!/bin/bash  

NAME="Zara Ali"  
readonly NAME  
NAME="Qadiri"  

#### RESULT:

>NAME: readonly variable  
---------------------------------  
#### Darbs ar speciāliem mainīgājiem  

#!/bin/bash

echo "File Name: $0"  
echo "First Parameter : $1"  
echo "Second Parameter : $2"  
echo "Quoted Values: $@"  
echo "Quoted Values: $*"  
echo "Total Number of Parameters : $#"  

#### RESULT:
> ./test_variables_3.sh Stas Kursish  
File Name : ./test_variables_3.sh  
First Parameter : Stas  
Second Parameter : Kursish  
Quoted Values: Stas Kursish  
Quoted Values: Stas Kursish  
Total Number of Parameters : 2 

---------------------------------  
  
#### Darbs ar mainīgo masīvu

#!/bin/bash

NAME[0]="Vitjok"  
NAME[1]="Mashka"  
NAME[2]="Zajchik343"  
NAME[3]="Popka"  
NAME[4]="Juljchik"  
echo "First Index: ${NAME[*]}"  
echo "Second Index: ${NAME[@]}"  

#### RESULT:
>First Index: Vitjok Mashka Zajchik343 Popka Juljchik
Second Index: Vitjok Mashka Zajchik343 Popka Juljchik
--------------------------------

#### Darbs ar operācijām

#### Arithmetic Operators

a=10  
b=20  
c=10  

echo $[ $a == $b ]  would return false (0)  
echo $[ $c != $a ]  would return false (0)  
echo $[ $a == $c ]  would return true (1)  
echo $[ $c != $b ]  would return true (1)  

#### Relational Operators

-eq - equal or not [ $a -eq $b ] - is not true  
-ne - not equal or is [ $a -ne $b ] - is true  
-gt - is one greater or is not [ $a -gt $b ] - is not true  
-lt - first operand is less than second [ $a -lt $b ] - is true  
-ge - first operand is greater or equal to second [ $a -ge $b ] - is not true  
-le - first operand is less or equal to second [ $a -le $b ] - is true  

#### Boolean Operators

! - logical negation. This inverts a true condition into false [ !false ] - is true  
-o - this is logical **OR**. If one of the operands is true, than the condition becomes true [ $a -lt 20 -o $b -gt 100] is true  
-a - this is logical **AND**. If both the operands, then the condition becomes true otherwise false [ $a -lt 20 -a $b -gt 100] is false  




test/history.txt
    1  VirtualBox --startvm XP
    2  quartus
    3  PS1="$ "
    4  whereis mysql
    5  mysql
    6  msql -u user
    7  mysql -u user
    8  mysql -u user -p
    9  history
   10  history > history.txt
   11  ls
   12  git clone https://github.com/rawkingo/EDIBO
   13  ls
   14  cd EDIBO/
   15  pwd
   16  ls
   17  nano git-upload
   18  git-upload
   19  /git-upload
   20  mkdir test
   21  pwd
   22  ls
   23  git-upload > test
   24  mv git-upload
   25  mv git-upload test
   26  ls
   27  cd test
   28  ls
   29  history > history.txt


SHELL/test.sh
!/bin/bash

NAME="Zara Ali"
echo $NAME

echo "File Name: $0"
echo "First Parameter : $1"
echo "Second Parameter : $2"
echo "Quoted Values: $@"
echo "Quoted Values: $*"
echo "Total Number of Parameters : $#"


SHELL/test_variables_3.sh
#!/bin/bash

echo "File Name: $0"
echo "First Parameter : $1"
echo "Second Parameter : $2"
echo "Quoted Values: $@"
echo "Quoted Values: $*"
echo "Total Number of Parameters : $#"



SHELL/test_variables_4.sh
#!/bin/bash

NAME[0]="Vitjok"
NAME[1]="Mashka"
NAME[2]="Zajchik343"
NAME[3]="Popka"
NAME[4]="Juljchik"
echo "First Index: ${NAME[*]}"
echo "Second Index: ${NAME[@]}"


Python/python_time.py
#!/usr/bin/python

print("Ievadiet skaitli ")
# a=2**2000000

#te ir tris darbibas - vertibas sagaidisana, vertibas parveidosana, pieskirsana
#argument = input()
#int(argument)
#a = int(argument)

#pildod int(input()) "bez izmeginajuma programma var vienkarsi izpildit ....
#tapec lai "nelidotu" mes izmantosim try ... except ... finaly konstrukcija

pazime = False
while not pazime:
#whle pazima = False:
#while paziime !=True: 

    try:
        a=int(input())
        pazime = True 
    except:
        print("Diemzel, cienijamais lietotaj, to kas ir ievadits nevar parveidot")
        print("Ludzu ievadi skaitli (0,1,2,3,4,5,6,7,8,9) velreiz")

#if (a == int): print ("a**100)
#ja loti gribas, tad var salidzinas type(a) == int -> rezultats bus True

if (a == 5):
    print(a**100)
    print("Aprekins ir gatavs")
print("Sis teksts atrodas arpus bloka - pierakstits bez"
      "atstarpem prieksa, tapec tas paradisies jebkura gadijuma")


Python/diary_20200813a.py
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license()" for more information.
>>> a=10
>>> int
<class 'int'>
>>> a
10
>>> 10 == int
False
>>> type(a) == int
True
>>> int(0)
0
>>> int(24)
24
>>> int(False)
0
>>> a=10
>>> if type(a) ==int:
	print("labi")
else:
	print("slikti")

	
labi
>>> a="10"
>>> 
>>> if type(a) == int:
	print("labi")
else:
	print("slikti")

	
slikti
>>> a=5.5
>>> type(a)
<class 'float'>
>>> if type(a) == int:
print("labi")
SyntaxError: expected an indented block
>>> if type(a) == int:
	print("labi")
elif type(a) == float:
        print("ari labi")
else:
	print("slikti")

	
ari labi
>>> a=10
>>> f type(a) == int:
	print("labi")
else:
	print("slikti")
	
SyntaxError: invalid syntax
>>> if type(a) == int:
	print("labi")
elif type(a) == float:
        print("ari labi")
else:
	print("slikti")

	
labi
>>> print ("aaa bbb ccc")
aaa bbb ccc
>>> print ("aaa \n bbb \n ccc")
aaa 
 bbb 
 ccc
>>> 


Python/diary_20200813b.py
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license()" for more information.
>>> a=10
>>> int
<class 'int'>
>>> a
10
>>> 10 == int
False
>>> type(a) == int
True
>>> int(0)
0
>>> int(24)
24
>>> int(False)
0
>>> a=10
>>> if type(a) ==int:
	print("labi")
else:
	print("slikti")

	
labi
>>> a="10"
>>> 
>>> if type(a) == int:
	print("labi")
else:
	print("slikti")

	
slikti
>>> a=5.5
>>> type(a)
<class 'float'>
>>> if type(a) == int:
print("labi")
SyntaxError: expected an indented block
>>> if type(a) == int:
	print("labi")
elif type(a) == float:
        print("ari labi")
else:
	print("slikti")

	
ari labi
>>> a=10
>>> f type(a) == int:
	print("labi")
else:
	print("slikti")
	
SyntaxError: invalid syntax
>>> if type(a) == int:
	print("labi")
elif type(a) == float:
        print("ari labi")
else:
	print("slikti")

	
labi
>>> print ("aaa bbb ccc")
aaa bbb ccc
>>> print ("aaa \n bbb \n ccc")
aaa 
 bbb 
 ccc
>>> fruit = 'banana'
>>> letter = fruit[1]
>>> print(letter)
a
>>> petter = fruit[4]
>>> print(petter)
n
>>> fruit = "carlson"
>>> fruit(len)
Traceback (most recent call last):
  File "<pyshell#41>", line 1, in <module>
    fruit(len)
TypeError: 'str' object is not callable
>>> len(fruit)
7
>>> fruit
'carlson'
>>> type(carlson)
Traceback (most recent call last):
  File "<pyshell#44>", line 1, in <module>
    type(carlson)
NameError: name 'carlson' is not defined
>>> type(fruit)
<class 'str'>
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
c
a
r
l
s
o
n
>>> while index > len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
>>> index = 7
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
>>> while index > len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
>>> while index > len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
>>> index = 6
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
n
>>> while index <len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
>>> while index > len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
>>> 
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index + 1

	
n
Traceback (most recent call last):
  File "<pyshell#77>", line 2, in <module>
    letter = fruit[index+6]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index+5]
	print(letter)
	index = index + 1

	
n
Traceback (most recent call last):
  File "<pyshell#79>", line 2, in <module>
    letter = fruit[index+5]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#81>", line 2, in <module>
    letter = fruit[index+6]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#83>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> while index > len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

>>> 
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#87>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index+1]
	print(letter)
	index = index - 1

	
c
Traceback (most recent call last):
  File "<pyshell#89>", line 2, in <module>
    letter = fruit[index+1]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 2

	
Traceback (most recent call last):
  File "<pyshell#91>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#93>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1while index < len(fruit):
		letter = fruit[index]
		print(letter)
		index = index - 1
		
SyntaxError: invalid syntax
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#96>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#98>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 7
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
>>> index = 6
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#104>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 5
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#107>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#109>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#112>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index
-8
>>> index = 1
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#116>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index
-8
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
c
a
r
l
s
o
n
>>> index
7
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#124>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index
-8
>>> fruit
'carlson'
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#128>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#131>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#133>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = 0
>>> while index > len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
>>> while index >len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index + 1

	
>>> 
>>> index
0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 7

	
n
Traceback (most recent call last):
  File "<pyshell#143>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#145>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#149>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#153>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index = 0
>>> letter = fruit[index-1]
>>> letter
'n'
>>> index
0
>>> int(letteR)
Traceback (most recent call last):
  File "<pyshell#158>", line 1, in <module>
    int(letteR)
NameError: name 'letteR' is not defined
>>> 
>>>  int(letter)
SyntaxError: unexpected indent
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#162>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index 7
SyntaxError: invalid syntax
>>> index = 7
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
>>> index = 6
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#170>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = 0
>>> letter = fruit[-7]
>>> letter
'c'
>>> letter = fruit[0]
>>> letter
'c'
>>> letter = fruit[-14]
Traceback (most recent call last):
  File "<pyshell#177>", line 1, in <module>
    letter = fruit[-14]
IndexError: string index out of range
>>> letter = fruit[8]
Traceback (most recent call last):
  File "<pyshell#178>", line 1, in <module>
    letter = fruit[8]
IndexError: string index out of range
>>> index
0
>>> while index > len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
>>> index =
SyntaxError: invalid syntax
>>> index
0
>>> index = -7
>>> while index > len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
>>> while index > len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index + 1

	
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
c
Traceback (most recent call last):
  File "<pyshell#190>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index
-8
>>> index = -7
>>> while index > len(fruit):
	letter = fruit[index]
	print(letter)
	index = index + 1

	
>>> index
-7
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#198>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = 0
>>> index = 6
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#203>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index
-8
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#207>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index = -6
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
c
Traceback (most recent call last):
  File "<pyshell#210>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = -6
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index + 1

	
c
a
r
l
s
o
n
c
a
r
l
s
o
>>> o
Traceback (most recent call last):
  File "<pyshell#215>", line 1, in <module>
    o
NameError: name 'o' is not defined
>>> index
7
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#219>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index = 0
>>> len(fruit)
7
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)

	
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c

c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
c
cTraceback (most recent call last):
  File "<pyshell#222>", line 3, in <module>
    print(letter)
KeyboardInterrupt
>>> while index < len(fruit):
	letter = fruit[index-7]
	print(letter)
	index = index + 1

	
c
a
r
l
s
o
n
>>> index
7
i
>>> index
7
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index+7]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#229>", line 2, in <module>
    letter = fruit[index+7]
IndexError: string index out of range
>>> index
0
>>> while index < len(fruit):
	letter = fruit[index+7]
	print(letter)
	index = index + 1

	
Traceback (most recent call last):
  File "<pyshell#232>", line 2, in <module>
    letter = fruit[index+7]
IndexError: string index out of range
>>> index
0
>>> while index < len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index + 1

	
n
Traceback (most recent call last):
  File "<pyshell#235>", line 2, in <module>
    letter = fruit[index+6]
IndexError: string index out of range
>>> index
1
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#239>", line 2, in <module>
    letter = fruit[index+6]
IndexError: string index out of range
>>> index
-14
>>> index = 0
>>> while index > len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index - 1

	
>>> index
0
>>> index = 12
>>> while index > len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#247>", line 2, in <module>
    letter = fruit[index+6]
IndexError: string index out of range
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#250>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index = 0
>>> letter = fruit[-1]
>>> letter
'n'
>>> letter = fruit[-2]
>>> letter
'o'
>>> index
0
>>> while index < len(fruit)-1:
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#258>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index = 0
>>> while index < len(fruit)-1:
	letter = fruit[index]
	print(letter)
	index = index - 1

	
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#261>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 6
>>> fruit
'carlson'
>>> fruit[6]
'n'
>>> fruit[index]
'n'
>>> while index <- len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
>>> index
6
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#270>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#273>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 0
>>> letter = fruit[index+6]
>>> fruit
'carlson'
>>> letter = fruit[index-1]
>>> fruit
'carlson'
>>> letter
'n'
>>> letter = fruit[index+6]
>>> letter
'n'
>>> while index < len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#283>", line 2, in <module>
    letter = fruit[index+6]
IndexError: string index out of range
>>> index
-14
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#287>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#290>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = -1
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#293>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 0
>>> while index < len(fruit)-1:
	letter = fruit[index]
	print(letter)
	index = index - 1

	
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#296>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index
-8
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#300>", line 2, in <module>
    letter = fruit[index+6]
IndexError: string index out of range
>>> index = 6
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#303>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index
-8
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#307>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = 0
>>> while index < len(fruit)-1:
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#311>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index+6]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#315>", line 2, in <module>
    letter = fruit[index+6]
IndexError: string index out of range
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
Traceback (most recent call last):
  File "<pyshell#317>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#320>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 6
>>> while index < len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#323>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index
-8
>>> index = 6
>>> while index <= len(fruit):
	letter = fruit[index]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#327>", line 2, in <module>
    letter = fruit[index]
IndexError: string index out of range
>>> index = 0
>>> while index <=len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1

	
n
o
s
l
r
a
c
Traceback (most recent call last):
  File "<pyshell#330>", line 2, in <module>
    letter = fruit[index-1]
IndexError: string index out of range
>>> index
-7
>>> index = 0
>>> while index < len(fruit):
	letter = fruit[index-1]
	print(letter)
	index = index - 1
